accounts-db/src/accounts.rs

```diff
@@ -719,11 +719,14 @@ impl Accounts {
                 TransactionExecutionResult::NotExecuted(_) => continue,
             };
 
-            let maybe_nonce = match (execution_status, &loaded_transaction.nonce) {
-                (Ok(_), _) => None, // Success, don't do any additional nonce processing
-                (Err(_), Some(nonce)) => {
-                    Some((nonce, true /* rollback */))
-                }
```

Comment:
Before, `maybe_nonce` would only be `Some(..)` if the tx execution status was failed and the transaction used a nonce. Also, `rollback` is always `true` in this case, it's never `false`. This means that `prepare_if_nonce_account` doesn't need to check `rollback` and doesn't need to check execution status if `maybe_nonce` is `Some(..)`